File: LoaneHpManagementFrontEnd/src/app/services/dealer-product.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { environment } from '../../environments/environment';
import { DealerProduct } from '../models/dealer-product.model';
import { ApiResponse, PagedResponse } from '../models/common.types';

@Injectable({
  providedIn: 'root'
})
export class DealerProductService {
  private apiUrl = `${environment.apiUrl}/dealer-products`;

  constructor(private http: HttpClient) {}

  getAllDealerProducts(
    page: number = 0,
    size: number = 10,
    sortBy: string = 'id'
  ): Observable<PagedResponse<DealerProduct>> {
    const params = { page: page.toString(), size: size.toString(), sortBy };
    return this.http.get<ApiResponse<PagedResponse<DealerProduct>>>(`${this.apiUrl}/getAll`, { params })
      .pipe(map(response => response.data));
  }

  getAllProductToSelect(searchTerm: string = ''): Observable<DealerProduct[]> {
    const params = { searchTerm };
    return this.http.get<DealerProduct[]>(`${this.apiUrl}/getForSelect`, { params });
  }

  getDealerProductById(id: number): Observable<DealerProduct> {
    return this.http.get<ApiResponse<DealerProduct>>(`${this.apiUrl}/list/${id}`)
      .pipe(map(response => response.data));
  }

  getDealerProductsByCifId(cifId: number): Observable<DealerProduct[]> {
    return this.http.get<ApiResponse<DealerProduct[]>>(`${this.apiUrl}/cif/${cifId}`)
      .pipe(map(response => response.data));
  }

  createDealerProduct(cifId: number, dealerProduct: DealerProduct): Observable<DealerProduct> {
    return this.http.post<ApiResponse<DealerProduct>>(`${this.apiUrl}/save/${cifId}`, dealerProduct)
      .pipe(map(response => response.data));
  }
 // In dealer-product.service.ts
// In dealer-product.service.ts
updateDealerProductStatus(id: number, status: string) {
  return this.http.put<any>(
    `${this.apiUrl}/${id}/status`, 
    { status }
  );
}

  updateDealerProduct(id: number, dealerProduct: DealerProduct): Observable<DealerProduct> {
    return this.http.put<ApiResponse<DealerProduct>>(`${this.apiUrl}/update/${id}`, dealerProduct)
      .pipe(map(response => response.data));
  }

  deleteDealerProduct(id: number): Observable<void> {
    return this.http.delete<ApiResponse<void>>(`${this.apiUrl}/delete/${id}`)
      .pipe(map(() => undefined));
  }
} 

File: LoneHPManagementBackend/src/main/java/com/prj/LoneHPManagement/Service/DealerProductService.java
package com.prj.LoneHPManagement.Service;

import com.prj.LoneHPManagement.model.entity.DealerProduct;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.Optional;

public interface DealerProductService {
    DealerProduct updateProductStatus(int id, String status);
    Page<DealerProduct> getAllDealerProducts(int page, int size, String sortBy);
    List<DealerProduct> getAllDealerProducts();
    List<DealerProduct> getProductsForSelect(String searchTerm);
    Optional<DealerProduct> getDealerProductById(int id);

//    List<DealerProduct> getDealerProductsByCompanyId(int companyId);

    DealerProduct createDealerProduct(int cifId, DealerProduct dealerProduct);

    DealerProduct updateDealerProduct(int id, DealerProduct dealerProductDetails);

    void deleteDealerProduct(int id);


}
